src/data/arena-tag.ts

```diff
@@ -919,6 +920,77 @@ class SafeguardTag extends ArenaTag {
   }
 }
 
+/**
+ * This arena tag facilitates the application of the move Imprison
+ * Imprison remains in effect as long as the source Pokemon is active and present on the field.
+ * Imprison will apply to any opposing Pokemon that switch onto the field as well.
+ */
+class ImprisonTag extends ArenaTrapTag {
+  private source: Pokemon;
+
+  constructor(sourceId: integer, side: ArenaTagSide) {
```

Comment:
NIT: No more `integer` please
```suggestion
  constructor(sourceId: number, side: ArenaTagSide) {
```